### PR TITLE
refactor(docs): 添加注册表文档

### PR DESCRIPTION
**Pull request by gene9831**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **Documentation**
  - Introduced `REGISTRY_README.md` to provide detailed documentation on the registry system for a designer tool.
  - Added examples and explanations for customizing and registering components.

- **New Features**
  - Defined structured type definitions (`VueComponent`, `OverwriteFn`, `MetaApp`, `AppRegistry`) to facilitate component registration and customization within a Vue component registry.

### CODE REVIEW

File: designer-demo/registry.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# 注册表
```
Comment (wenmine) [inline designer-demo/registry.md]: 感觉直接叫 `registry.md` 不大合适，能不能改成  `REGISTRY.README.md`

File: designer-demo/registry.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  component: VueComponent
+  // 组件暴露的api接口
+  apis?: Record<string, Function>
+  // component的配置项
+  options?: Record<string, any>
+  // 布局元应用。如果当前元应用需要定制布局，可以抽取一层布局元应用
+  layout?: MetaApp
+  // 元应用生命周期
+  lifeCycle?: Record<string, Function>
+  overwrite?: {
```
Comment (wenmine) [inline designer-demo/registry.md]: overwrite里的和上面的区别能不能也加个注释，比如说overwrite里的liftCycle和19行的有啥区别

File: designer-demo/registry.md (hunk 1 above)
Comment (gene9831) [reply designer-demo/registry.md]: 已重命名

File: designer-demo/REGISTRY_README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  // 组件暴露的api接口
+  apis?: Record<string, Function>
+  // component的配置项
+  options?: Record<string, any>
+  // 布局元应用。如果当前元应用需要定制布局，可以抽取一层布局元应用
+  layout?: MetaApp
+  // 元应用生命周期。此处写的代码会直接插入到应用中
+  lifeCycle?: Record<string, Function>
+  // 覆盖逻辑
+  overwrite?: {
```
Comment (zzcr) [inline designer-demo/REGISTRY_README.md:23]: 这里可以解释一下：只有元应用和元服务可以配置overwrite，插件工具栏大颗粒度得不能配置overwrite

File: designer-demo/REGISTRY_README.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+interface MetaApp {
+  id: string
+  component: VueComponent
+  // 组件暴露的api接口
+  apis?: Record<string, Function>
+  // component的配置项
+  options?: Record<string, any>
+  // 布局元应用。如果当前元应用需要定制布局，可以抽取一层布局元应用
+  layout?: MetaApp
```
Comment (zzcr) [inline designer-demo/REGISTRY_README.md:19]: 这里需要评审一下：目前根节点得layout和插件layout有所区别

File: designer-demo/registry.md (hunk 2 above)
Comment (gene9831) [reply designer-demo/registry.md]: 已添加

**Pull request merged**